jupyter_sphinx/ast.py

```diff
@@ -358,3 +400,50 @@ def get_widgets(notebook):
         # Don't catch KeyError, as it's a bug if 'widgets' does
         # not contain 'WIDGET_STATE_MIMETYPE'
         return None
+
+
+class CellOutputsToNodes(SphinxTransform):
+    """Use the builder context to transform a CellOutputNode into Sphinx nodes."""
+
+    default_priority = 700
+
+    def apply(self):
+        thebe_config = self.config.jupyter_sphinx_thebelab_config
+
+        for cell_node in self.document.traverse(JupyterCellNode):
+            output_bundle_node = _return_first_node_type(
+                cell_node, CellOutputBundleNode
+            )
+            # Create doctree nodes for cell outputs.
+            output_nodes = cell_output_to_nodes(
+                output_bundle_node.outputs,
+                self.config.jupyter_execute_data_priority,
+                bool(cell_node.attributes["stderr"]),
+                sphinx_abs_dir(self.env),
+                thebe_config,
+            )
+            # Remove the outputbundlenode and we'll attach the outputs next
+            attach_outputs(output_nodes, cell_node, thebe_config, cell_node.cm_language)
+
+        # Image collect extra nodes from cell outputs that we need to process
+        for node in self.document.traverse(image):
+            # If the image node has `candidates` then it's already been processed
+            # as in-line markdown, so skip it
```

Comment:
Not sure what you mean here with "inline markdown". Can you please explain?

Reply:
TBH I am not sure, I will try and spend a bit of time early this week to look through this again

Reply:
From the processing around I thought you meant that the image what in the document source already. It's just the "inline markdown" that got me confused.

Reply:
yeah I think I probably just used "markdown" because it was coming from myst-markdown, but the overall principle applies, I'll remove the use of "markdown"